paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/ContinuousFileSplitEnumerator.java

```diff
@@ -210,6 +211,8 @@ protected void processDiscoveredSplits(
                 LOG.debug("Catching EndOfStreamException, the stream is finished.");
                 finished = true;
                 assignSplits();
+            } else if (error instanceof OutOfRangeException) {
+                throw (RuntimeException) error;
             } else {
                 LOG.error("Failed to enumerate files", error);
```

Comment:
Maybe we need to throw all exceptions here?

Reply:
I agree with that. Thank you